vllm/worker/worker.py

```diff
@@ -208,24 +225,29 @@ def _prepare_inputs(
         # Add generation tokens.
         max_context_len = 0
         max_num_blocks_per_seq = 0
-        context_lens: List[int] = []
         generation_block_tables: List[List[int]] = []
-        max_seq_len = max(prompt_lens) if prompt_lens else 1
+        max_seq_len = max(subquery_lens) if subquery_lens else 1
         for i, seq_group_metadata in enumerate(seq_group_metadata_list):
             if seq_group_metadata.is_prompt:
                 # We need to do this in this loop as we need to know max_seq_len
                 assert len(
                     seq_ids) == 1, "Prompt input should have only one seq."
                 sampling_params = seq_group_metadata.sampling_params
                 assert len(prompt_lens) == len(seq_group_metadata_list)
-                prompt_len = prompt_lens[i]
+                # here is the subprompt len
+                prompt_len = subquery_lens[i]
                 if sampling_params.prompt_logprobs is not None:
                     selected_token_indices.extend(
                         range(selected_token_start_idx,
                               selected_token_start_idx + prompt_len - 1))
                 selected_token_indices.append(selected_token_start_idx +
                                               prompt_len - 1)
                 selected_token_start_idx += max_seq_len
+
+                # set the prefix state
```

Comment:
when tp>1, seq_group_metadata.prefix here is copied  by ray workers, so  `on_gpu=true` won't work on multi gpus.

Reply:
cc. @DouHappy

Reply:
Thank you for your reply. This help me a lot.